python/pyarrow/types.pxi

```diff
@@ -2386,25 +2426,102 @@ def struct(fields):
     return pyarrow_wrap_data_type(struct_type)
 
 
+cdef _extract_union_params(children_fields, type_codes,
+                           vector[shared_ptr[CField]]* c_fields,
+                           vector[int8_t]* c_type_codes):
+    cdef:
+        Field child_field
+
+    for child_field in children_fields:
+        c_fields[0].push_back(child_field.sp_field)
+
+    if type_codes is not None:
+        if len(type_codes) != <Py_ssize_t>(c_fields.size()):
+            raise ValueError("type_codes should have the same length "
+                             "as fields")
+        for code in type_codes:
+            c_type_codes[0].push_back(code)
+    else:
+        c_type_codes[0] = range(c_fields.size())
+
+
+def sparse_union(children_fields, type_codes=None):
+    """
+    Create SparseUnionType from children fields.
+
+    A union is defined by an ordered sequence of child types; each slot in
+    the union can have a value chosen from these types.
+
+    Parameters
+    ----------
+    fields : sequence of Field values
```

Comment:
This should be `children_fields : ...` to match the signature, yeah? (or `child_fields` if we go that naming convention)

Reply:
What is better? `children_fields` or `child_fields`?

Reply:
`child_fields` sounds better to me and matches what's inside the function